channel-sdk/build.gradle

```diff
@@ -32,118 +35,125 @@ android {
 		}
 	}
 	compileOptions {
-		sourceCompatibility JavaVersion.VERSION_1_8
-		targetCompatibility JavaVersion.VERSION_1_8
+		sourceCompatibility JavaVersion.VERSION_11
+		targetCompatibility JavaVersion.VERSION_11
 	}
-	task sdkDocumentation(type: org.jetbrains.dokka.gradle.DokkaAndroidTask) {
-		outputFormat = 'html'
-		outputDirectory = "../javadoc"
-		includeNonPublic = false
-		skipEmptyPackages = true
-
-		// To make sure we didn't miss anything
-		reportUndocumented = true
-
-		packageOptions {
-			prefix = "android"
-			suppress = true
-		}
-		packageOptions {
-			prefix = "com.zello.channel.sdk.platform"
-			suppress = true
-		}
+	kotlinOptions {
+		jvmTarget = JavaVersion.VERSION_11
+		freeCompilerArgs += "-Xopt-in=kotlin.RequiresOptIn"
+		freeCompilerArgs += "-Xjvm-default=enable"
 	}
 
-	task sourcesJar(type: Jar) {
-		archiveClassifier = 'sources'
-		from android.sourceSets.main.java.srcDirs
-	}
-	task javadocJar(type: Jar, dependsOn: sdkDocumentation) {
-		archiveClassifier = 'javadoc'
-		from sdkDocumentation.outputDirectory
+	// Library for testing
+	useLibrary 'android.test.mock'
+	buildToolsVersion '32.0.0'
+}
+
+task sdkDocumentation(type: DokkaTask) {
+	dokkaSourceSets {
+		named("main") {
+			perPackageOption {
+				matchingRegex.set("com.zello.channel.sdk.platform")
+				suppress.set(true)
+			}
```

Comment:
Package suppress option needs to go before the flags or else it's ignored for some reason.

Reply:
Yep. The only way I was able to find.